google_analytics/notebooks/load.ipynb

```diff
@@ -34,24 +34,189 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 67,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Out of 263866 single visits, 1107 ended up in a purchase\n"
+     ]
+    }
+   ],
+   "source": [
+    "# But maybe some of those guys actually bought something in that single visit.\n",
+    "single_visits = x_train[x_train[\"fullVisitorId\"].isin(single_visit_ids)]\n",
+    "num_purchases = len(single_visits[single_visits[\"transactionRevenue\"] > 0][\"transactionRevenue\"])\n",
+    "print(\"Out of {} single visits, {} ended up in a purchase\".format(len(single_visit_ids), num_purchases))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 68,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "We will therefore choose to ignore 262759 users\n"
+     ]
+    }
+   ],
+   "source": [
+    "to_ignore = set(single_visit_ids) - set(single_visits[single_visits[\"transactionRevenue\"] > 0][\"fullVisitorId\"])\n",
+    "print(\"We will therefore choose to ignore {} users\".format(len(to_ignore)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Is this assumption justified?\n",
+    "\n",
+    "How reliable is this assumption? Are these people that we choose to ignore indeed buying nothing? Let's find out.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 2290 people appearing both in the training and the test periods\n",
+      "We will ignore 2037 out of those 2290 people\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We will only focus on people who exist in both datasets\n",
+    "common_ids = set(y_train[\"fullVisitorId\"]).intersection(x_train[\"fullVisitorId\"])\n",
+    "print(\"There are {} people appearing both in the training and the test periods\".format(len(common_ids)))\n",
     "\n",
-    "# preprocess_and_save(\"../data\", nrows_train=100000, nrows_test=10000)\n",
-    "train, test = process(train, test)"
+    "y_train = y_train[y_train[\"fullVisitorId\"].isin(common_ids)]\n",
+    "ignored = y_train[y_train[\"fullVisitorId\"].isin(to_ignore)]\n",
+    "print(\"We will ignore {} out of those {} people\".format(len(ignored), len(common_ids)))"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 88,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "We made 196 mistakes (9.62% of cases) leading to loss = 5.4433800738203315\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "predicted_target = [0] * len(common_ids)\n",
```

Comment:
Were we set the predicted_target to zero for all common_ids. This means that also for people who are visiting and buying on a regular basis in y_train, we predict zero. I am correct with what I am saying here? Since I think this is not what we want?

Reply:
My intention was to predict 0 for those that should be ignored and compute the loss incurred by them alone. I’m not sure I did it correctly (hard to see the code on the phone :P)

Reply:
I will check it to be sure.